frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,892 @@
+
+						// Make payout to beneficiary.
+						let _ = T::Currency::transfer(
+							&child_bounty_account,
+							beneficiary,
+							payout,
+							AllowDeath,
+						);
+
+						// Trigger the ChildBountyClaimed event.
+						Self::deposit_event(Event::<T>::ChildBountyClaimed {
+							index: parent_bounty_id,
+							child_index: child_bounty_id,
+							payout,
+							beneficiary: beneficiary.clone(),
+						});
+
+						// Update the active child-bounty tracking count.
+						<ParentChildBounties<T>>::mutate(parent_bounty_id, |count| {
+							*count = count.saturating_sub(1)
+						});
+
+						// Remove the child-bounty description.
+						<ChildBountyDescriptions<T>>::remove(child_bounty_id);
+
+						// Remove the child-bounty instance from DB.
+						*maybe_child_bounty = None;
+
+						Ok(())
+					} else {
+						Err(BountiesError::<T>::UnexpectedStatus.into())
+					}
+				},
+			)
+		}
+
+		/// Cancel a proposed or active child-bounty.
+		/// Child-bounty account funds are transferred to parent bounty account.
+		/// The child-bounty curator deposit may be unreserved if possible.
+		///
+		/// The dispatch origin for this call must be
+		/// either parent curator or `T::RejectOrigin`.
+		///
+		/// If the state of child-bounty is `Active`,
+		/// curator deposit is unreserved.
+		///
+		/// If the state of child-bounty is `PendingPayout`,
+		/// call fails & returns `PendingPayout` error.
+		///
+		/// For the origin other than T::RejectOrigin,
+		/// parent bounty must be in active state,
+		/// for this child-bounty call to work. For origin
+		/// T::RejectOrigin execution is forced.
+		///
+		/// Instance of child-bounty is removed from DB
+		/// on successful call completion.
+		///
+		/// - `parent_bounty_id`: Index of parent bounty.
+		/// - `child_bounty_id`: Index of child bounty.
+		#[pallet::weight(<T as Config>::WeightInfo::close_child_bounty_added()
+			.max(<T as Config>::WeightInfo::close_child_bounty_active()))]
+		pub fn close_child_bounty(
+			origin: OriginFor<T>,
+			#[pallet::compact] parent_bounty_id: BountyIndex,
+			#[pallet::compact] child_bounty_id: BountyIndex,
+		) -> DispatchResult {
+			let maybe_sender = ensure_signed(origin.clone())
+				.map(Some)
+				.or_else(|_| T::RejectOrigin::ensure_origin(origin).map(|_| None))?;
+
+			// Ensure parent bounty exist, get parent curator.
+			let (parent_curator, _) = Self::ensure_bounty_active(parent_bounty_id)?;
+
+			ensure!(maybe_sender.map_or(true, |sender| parent_curator == sender), BadOrigin);
+
+			Self::impl_close_child_bounty(parent_bounty_id, child_bounty_id)?;
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	/// The account ID of a child-bounty account.
+	pub fn child_bounty_account_id(id: BountyIndex) -> T::AccountId {
+		// only use two byte prefix to support 16 byte account id (used by test)
+		// "modl" ++ "py/trsry" ++ "bt" is 14 bytes, and two bytes remaining for bounty index
+		T::PalletId::get().into_sub_account(("cb", id))
```

Comment:
what does it mean here to use 2 bytes for the bounty index, when it is a 4 byte number (u32)?

Reply:
I think that is for the prefix and not the index. This function is taken from the parent pallet, but the prefix is changed to have different AccountId when the index of parent and child is same.

Reply:
I think then the code comment should say this ^^, not the original one copy-pasted.